app/build.gradle

```diff
@@ -118,11 +118,22 @@ dependencies {
   androidTestImplementation "org.mockito:mockito-android:2.7.22"
 
   // Leak canary
-  debugImplementation 'com.squareup.leakcanary:leakcanary-android:1.5.4'
+  implementation 'com.squareup.leakcanary:leakcanary-android:1.5.4'
   releaseImplementation 'com.squareup.leakcanary:leakcanary-android-no-op:1.5.4'
```

Comment:
@brijeshshah13 @rashiq @mhutti1 @abdulwd 
 Can you please help here ? 
I think the reason why the build fails is the fact that i have declared leak canary dependency in the debug, test and release versions only. There are total 5 build variants. 
I am trying to declare it for them also but it just won't work.
for eg. for ```mock_network``` build variant
``` Groovy
mock_networkImplementation 'com.squareup.leakcanary:leakcanary-android:1.5.4' 
```
It doesn't work. Any suggestions?

The final aim is to add 
``` Groovy
'com.squareup.leakcanary:leakcanary-android:1.5.4'
``` 
dependency to every build variant except Release and test.